### ISSUE TITLE
`runRoute` doesn't seem to handle `Capture` properly

### ISSUE THREAD

**Issue by tysonzero** [dmjio/miso#374]
The following code gives `404 Not Found` when accessing it from `localhost:3000/foo/1/`, and I am not sure why, `localhost:3000/` correctly gives `home` and `localhost:3000/bar/` correctly gives `404 Not Found`.

The equivalent server side routing works just fine (when using `miso` rather than `startApp`), which on a side note actually lead to a runtime crash with a really difficult to debug error message about `undefined` not having a `length`.

```haskell
newtype AppState = AppState URI
    deriving (Eq, Show)

instance HasURI AppState where
    lensURI f (AppState uri) = AppState <$> f uri

initialState :: URI -> AppState
initialState = AppState

data Action
    = NoOp
    | ChangeURI URI
    | HandleURI URI

main :: IO ()
main = do
    currentURI <- getCurrentURI
    startApp App
        { model = initialState currentURI
        , update = handler
        , view = mainView
        , initialAction = NoOp
        , events = defaultEvents
        , mountPoint = Nothing
        , subs = [uriSub HandleURI]
        }

mainView :: AppState -> View AppAction
mainView m = case runRoute (Proxy @Routes) viewTree m of
    Left _ -> text "404 Not Found"
    Right v -> v
  where
    viewTree = homeView :<|> fooView

type Routes = HomeRoute
         :<|> FooRoute

type HomeRoute = View AppAction
type FooRoute = "foo" :> Capture "id" Int :> View AppAction

homeView :: s -> View a
homeView _ _ = div_ [] [text "home"]

fooView :: Int -> s -> View a
fooView x _ = div_ [] [text . pack $ "foo: " ++ show x]
```

**Comment by tysonzero** [comment]
Turns out it is not about `Capture`, it's about trailing slashes. The server/servant treats as optional, but the client/`runRoute` treats them as invalid, so if you put a trailing `/` in the URL you will crash the app with a hard to debug error message. IMO `runRoute` should be brought in line with servant.

**Comment by tysonzero** [comment]
On a side note, how difficult would it be to have client/server desync be handled by spitting out a warning showing the difference between the client and the server, and then throwing out the server rendered tree? Currently they are pretty hard to debug, and in prod i'd much rather lose server side rendering then lose the whole site.

**Comment by dmjio** [comment]
Can you paste your `update` function here for context?

**Comment by tysonzero** [comment]
```
handler :: Action -> AppState -> Effect Action AppState
handler NoOp m = noEff m
handler (ChangeURI u) m = m <# do
    pushURI u
    pure NoOp
handler (HandleURI u) _ = noEff $ AppState u
```

**Comment by FPtje** [comment]
I cannot reproduce the issue with latest Miso. Tested with my isomorphic example by opening both `http://localhost:3003/flipped` and `http://localhost:3003/flipped/`.

Was this fixed in 21e16ddcf6e4c427bd0a9dfe610ce328a828477d?

**Comment by dmjio** [comment]
Closing for now, please re-open if the problem persists.
